core/src/main/java/org/apache/accumulo/core/metadata/StoredTabletFile.java

```diff
@@ -291,9 +291,41 @@ public TabletFileCq(Path path, Range range) {
     }
   }
 
-  private static class TabletFileCqMetadataGson {
+  static class TabletFileCqMetadataGson {
     private String path;
     private byte[] startRow;
     private byte[] endRow;
+
+    TabletFileCqMetadataGson() {}
+
+    TabletFileCqMetadataGson(AbstractTabletFile<?> atf) {
+      path = Objects.requireNonNull(atf.path.toString());
+      startRow = encodeRow(atf.range.getStartKey());
+      endRow = encodeRow(atf.range.getEndKey());
+    }
+
+    ReferencedTabletFile toReferencedTabletFile() {
+      return new ReferencedTabletFile(new Path(URI.create(path)),
+          new Range(decodeRow(startRow), true, decodeRow(endRow), false));
+    }
+
+    StoredTabletFile toStoredTabletFile() {
+      return new StoredTabletFile(new TabletFileCq(new Path(URI.create(path)),
+          new Range(decodeRow(startRow), true, decodeRow(endRow), false)));
+    }
+  }
+
+  static StoredTabletFile fomTabletFileCqMetadataGson(String serialized) {
+    return new StoredTabletFile(deserialize(serialized));
+  }
+
+  TabletFileCqMetadataGson toTabletFileCqMetadataGson() {
+    requireKeyExtentDataRange(range);
+    final TabletFileCqMetadataGson metadata = new TabletFileCqMetadataGson();
+    metadata.path = Objects.requireNonNull(path).toString();
+    metadata.startRow = encodeRow(range.getStartKey());
+    metadata.endRow = encodeRow(range.getEndKey());
+    return metadata;
```

Comment:
May be able to do the following not sure.  Also the requireKeyExtentDataRange() call may not be needed because StoredTableFile is immutable and the range should be verified already, but not completely sure about this.

```suggestion
  TabletFileCqMetadataGson toTabletFileCqMetadataGson() {
       return new TabletFileCqMetadataGson(this);
```

Reply:
Good call, I ended up just removing this method entirely in acfa49f and used the TabletFileCqMetadataGson constructor in both cases in CompactionMetadata.